sources/packages/api/src/route-controllers/supporting-user/supporting-user.aest.controller.ts

```diff
@@ -33,4 +40,49 @@ export class AESTSupportingUserController {
         } as ApplicationSupportingUsersDTO),
     );
   }
+
+  /**
+   * Get supporting user formName and the form data
+   * with the supportingUserId
+   * @param supportingUserId supporting user id.
+   * @param applicationId application id.
+   * @returns supporting user form data and details.
+   */
+  @Get(":supportingUserId/application/:applicationId/aest")
+  async getSupportingUserFormDetails(
+    @Param("supportingUserId") supportingUserId: number,
+    @Param("applicationId") applicationId: number,
+  ): Promise<SupportingUserFormData> {
+    const supportingUserForms =
+      await this.applicationService.getSupportingUserFormName(applicationId);
+
+    if (!supportingUserForms) {
+      throw new NotFoundException(
+        `Application ${applicationId} not found or Program year associated with the application not found`,
+      );
+    }
+    const supportingUserForApplication =
+      await this.supportingUserService.getSupportingUsersDetails(
+        supportingUserId,
+      );
+    if (!supportingUserForApplication) {
+      throw new NotFoundException(
+        `Supporting user ${supportingUserId} details not found or Supporting user has  not submitted the form`,
+      );
+    }
+    return {
+      formName: getSupportingUserForm(
+        supportingUserForApplication.supportingUserType,
+        supportingUserForms.programYear,
+      ),
+      supportingData: supportingUserForApplication.supportingData,
+      contactInfo: supportingUserForApplication.contactInfo,
+      sin: supportingUserForApplication.sin,
+      birthDate: supportingUserForApplication.birthDate,
+      gender: supportingUserForApplication.gender,
+      email: supportingUserForApplication.user.email,
```

Comment:
since contactInfo, sin, birthDate and gender are optional attributes in your model definition above, it will be good to also indicate them here too. E.g supportingUserForApplication?.contactInfo, supportingUserForApplication?.sin, etc.

Reply:
Hi @henryoforeh-aot, can you please review the documentation below?
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining

The below code suggests that _supportingUserForApplication_ is possibly null.
```
 supportingUserForApplication?.contactInfo
```

Reply:
Based on the response from @ann-aot, my comment here is no longer valid.
![annresponse](https://user-images.githubusercontent.com/72251620/155780880-c7e759ef-881c-4b72-9374-90fd1d4f8e9f.jpg)

Reply:
@henryoforeh-aot, even with @ann-aot comments, the first comment was already invalid. I believe that we talked about it before and I just would like to avoid the same conversation again. Please let me know if we need discuss it further?